File: ARC/ARC124/_a.cpp
#include <algorithm>
#include <iostream>
#include <iomanip>
#include <limits.h>
#include <map>
#include <math.h>
#include <numeric>
#include <queue>
#include <set>
#include <sstream>
#include <string>
#include <utility>
#include <vector>
#include <stack>

using namespace std;
#define rep(i, n) for (int i = 0; i < n; i++)
#define rep1(i, n) for (int i = 1; i < n + 1; i++)
#define all(A) A.begin(), A.end()
#define itr(A, l, r) A.begin() + l, A.begin() + r
#define debug(var) cout << #var << " = " << var << endl;
typedef long long ll;

#include <atcoder/modint>
using namespace atcoder;
using mint = modint998244353;

int main(void)
{
    int n, K;
    cin >> n >> K;
    vector<mint> dp(n);
    vector<bool> free(n, true);
    rep(i, K)
    {
        char c;
        int k;
        cin >> c >> k;
        k--;
        free[k] = false;
        rep(j, n)
        {
            if (c == 'R' && j <= k)
            {
                dp[j]++;
            }
            if (c == 'L' && k <= j)
            {
                dp[j]++;
            }
        }
    }
    mint ans = 1;
    rep(i, n)
    {
        if (free[i])
        {
            ans *= dp[i];
        }
    }
    cout << ans.val() << endl;
}

File: ARC/ARC124/_a.py
n, K = list(map(int, input().split(' ')))
imos = [0] * (n+1)
free = [True] * n
for i in range(K):
    c, k = input().split(' ')
    k = int(k)
    k -= 1
    free[k] = False
    if c == 'R':
        imos[0] += 1
        imos[k+1] -= 1
    if c == 'L':
        imos[k] += 1
        imos[n] -= 1
dp = [0] * (n+1)
for i in range(n):
    dp[i] = imos[i]
    if i > 0:
        dp[i] += dp[i-1]
ans = 1
for i in range(n):
    if free[i]:
        ans *= dp[i]
        ans %= 998244353
print(ans)
